docs/concepts/composition.md

```diff
@@ -0,0 +1,134 @@
+Similarly, when you work on hobby projects it's likely that you use a similar set of tools from one project to the next.
+With other developer environment solutions it's common to need to need to recreate your developer environment from scratch for each new project.
+
+With Flox you can define a toolchain or service once and _compose_ it with others to create a composed environment.
+Put another way, **with Flox you can create developer environments from reusable building blocks**.
+Reuse and composition are two features that have long been a Holy Grail for developer environments, and Flox makes these features friendly and easily accessible.
+
+## Building blocks
+
+Environments are the building blocks from which a "composed" environment is created.
+In this composition there is a hierarchy consisting of a "composing" environment and a list of "included" environments that it treats as dependencies.
+The manifests of the included and composing environments are merged and then the resulting "merged" manifest is locked and built.
+
+```d2 scale="0.9"
+composer: "Composing\nenvironment"
+
+includes: Included environments {
+  envA: Env A
+  envB: Env B
+  envC: Env C
+  direction: down
+  envA -> envB: Overridden by
+  envB -> envC: Overridden by
+}
+
+composer -> includes: Overrides
+```
+
+## Including an environment
+
+The included environments are declared in your manifest in the `include` table under the `environments` array.
+Each entry in this array is an "include descriptor", which specifies where to find the environment.
+
+```text
+IncludedEnvironments ::= [IncludeDescriptor]
+IncludeDescriptor    ::= PathEnvDescriptor | RemoteEnvDescriptor
+PathEnvDescriptor    ::= {
+  dir  = <PATH>
+, name = null | <STRING>
+}
+RemoteEnvDescriptor  ::= {
+  remote = <owner/name>
+, name   = null | <STRING>
+}
+```
+
+Just like you would use the `flox activate --dir` flag to specify an environment to activate by its path, you use the `dir` field to specify the path to an environment to include.
+Similarly, you use the `remote` field to specify the owner and name of an environment on FloxHub to include e.g. `"flox/some_env"`.
+Every environment has a name built-in, but sometimes there may be name conflicts, or you may just want to provide a different name.
+You can do so with the `name` field.
+
+An example `include` section is shown below:
+
+```toml
+[include]
+environments = [
+  { dir = "../myenv" },
+  { remote = "flox/some_env" },
+  # Override the name of this environment
+  { dir = "../other_env", name = "other" }
+]
+```
+
+The order in which the included environments are listed matters.
+The environments specified later in the list will override those earlier in the list.
+
+It is possible to push a composed environment to FloxHub, but not if it includes environments that exist locally e.g. environments that are specified with the `dir` field.
+
+## Merging process
```

Comment:
(non-blocking) There's a good chance that this is going to get out-of-step with the `manifest.toml` man page and the implementation. I think we should consider consolidating and linking at some point.

Reply:
I think that's a post-release thing. Part of the motivation for getting these in early is so that we can point at them for the bug bash, whereas it's not straightforward to get the manpages in early.